src/SldStyleParser.ts

```diff
@@ -292,188 +284,203 @@ export class SldStyleParser implements StyleParser<string> {
+   *
+   * @param sldRule The SLD Rule
+   * @return The geostyler-style Symbolizer array
+   */
+  getSymbolizersFromRule(sldRule: any[]): Symbolizer[] {
+    const symbolizers: Symbolizer[] = sldRule
+      .filter(isSymbolizer)
+      .map((sldSymbolizer: any) => {
+        const sldSymbolizerName: string = Object.keys(sldSymbolizer)[0];
+        switch (sldSymbolizerName) {
+          case 'PointSymbolizer':
+            return this.getPointSymbolizerFromSldSymbolizer(sldSymbolizer.PointSymbolizer);
+          case 'LineSymbolizer':
+            return this.getLineSymbolizerFromSldSymbolizer(sldSymbolizer.LineSymbolizer);
+          case 'TextSymbolizer':
+            return this.getTextSymbolizerFromSldSymbolizer(sldSymbolizer.TextSymbolizer);
+          case 'PolygonSymbolizer':
+            return this.getFillSymbolizerFromSldSymbolizer(sldSymbolizer.PolygonSymbolizer);
+          case 'RasterSymbolizer':
+            return this.getRasterSymbolizerFromSldSymbolizer(sldSymbolizer.RasterSymbolizer);
+          default:
+            throw new Error('Failed to parse SymbolizerKind from SldRule');
+        }
+      });
+
+    return symbolizers;
   }
 
   /**
-   * Creates a GeoStyler-Style Filter from a given operator name and the js
-   * SLD object representation (created with xml2js) of the SLD Filter.
+   * Creates a geostyler-style Filter from a given operator name and the js
+   * SLD object representation (created with fast-xml-parser) of the SLD Filter.
    *
-   * @param {string} sldOperatorName The Name of the SLD Filter Operator
-   * @param {object} sldFilter The SLD Filter
-   * @return {Filter} The GeoStyler-Style Filter
+   * @param sldOperatorName The Name of the SLD Filter Operator
+   * @param sldFilter The SLD Filter
+   * @return The geostyler-style Filter
    */
-  getFilterFromOperatorAndComparison(sldOperatorName: string, sldFilter: any): Filter {
-    let filter: Filter;
+  getFilterFromOperatorAndComparison(sldOperatorName: string, sldFilter: any): GsFilter {
+    let filter: GsFilter;
 
     // we have to first check for PropertyIsBetween,
     // since it is also a comparisonOperator. But it
     // needs to be treated differently.
     if (sldOperatorName === 'PropertyIsBetween') {
-      // TODO PropertyIsBetween spec allows more than just a
-      //      PropertyName as its first argument.
-      const propertyName = sldFilter.PropertyName[0]._;
-      const lower = parseFloat(sldFilter.LowerBoundary[0].Literal[0]._);
-      const upper = parseFloat(sldFilter.UpperBoundary[0].Literal[0]._);
+      // TODO: PropertyIsBetween spec allows more than just a PropertyName as its first argument.
+      const propertyName = get(sldFilter, 'PropertyIsBetween.PropertyName.#text');
+      const lower = Number(get(sldFilter, 'PropertyIsBetween.LowerBoundary.Literal.#text'));
+      const upper = Number(get(sldFilter, 'PropertyIsBetween.UpperBoundary.Literal.#text'));
```

Comment:
Why was the handling of `shouldParseFloat` removed?

Reply:
See CHANGELOG.md